tensorboard/webapp/notification_center/_data_source/notification_center_data_source_test.ts

```diff
@@ -85,4 +88,38 @@ describe('TBNotificationCenterDataSource test', () => {
     expect(resultSpy).not.toHaveBeenCalled();
     expect(errorSpy).toHaveBeenCalledWith(httpErrorResponse);
   });
+
+  describe('LastReadTimestamp in local storage test', () => {
+    let TEST_TIME_STAMP_STRING = '-1';
+    const INIT_LAST_READ_TIMES_TAMP_STRING = '1235813';
+    const mocklocalStorage: {[key: string]: string} = {
+      [NOTIFICATION_LAST_READ_TIME_KEY]: INIT_LAST_READ_TIMES_TAMP_STRING,
+    };
+
+    beforeEach(async () => {
+      spyOn(window.localStorage, 'setItem').and.callFake(
+        (key: string, value: string): string => {
+          TEST_TIME_STAMP_STRING = value;
+          return (mocklocalStorage[key] = value);
+        }
+      );
+      spyOn(localStorage, 'getItem').and.callFake((key: string):
+        | string
+        | null => {
+        return mocklocalStorage[key] || null;
+      });
+    });
+
+    it('updates last read timestamp in local storge', () => {
```

Comment:
there should be an assertion of the `TEST_TIME_STAMP_STRING`, not assertion on the fake implementation you've made (`getItem`).